lgc/interface/lgc/Pipeline.h

```diff
@@ -181,11 +181,12 @@ enum class ResourceNodeType : unsigned {
   DescriptorBuffer,          ///< Generic descriptor: buffer, including uniform buffer and shader storage buffer
   DescriptorTableVaPtr,      ///< Descriptor table VA pointer
   IndirectUserDataVaPtr,     ///< Indirect user data VA pointer
-  PushConst,                 ///< Push constant
+  PushConst,                 ///< Push constant; only a single PushConst in the root table is allowed
```

Comment:
I believe the push constants are inline in the root user data, rather than there being a pointer to them in the root user data.

Reply:
You're right, push constants are always in the user data table (the Pal::ICmdBuffer::CmdSetUserData thing), which means they're in user data registers and/or in the spill table (the spill table, if it exists, is equal to the user data table).

There seems to be an inefficiency here in that we seem to be either loading all push constants from user data or loading all push constants from the spill table, but if we can fit push constants partially into user data, we don't make use of that. (At least it seems that in this case, we aren't adding the push constants to the PAL metadata? So at least we're not penalizing wave launch.) Addressing this could be separate from this particular pull request, of course.

Back to the InlineBuffer -- the driver itself never defines InlineBuffer as part of the pipeline layout, right? After all, the enum didn't exist before this.

Does the driver ever define PushConst as a non-root node? It seems to me that it shouldn't, at least not for Vulkan... in which case we could perhaps reduce complexity further by just sticking an assertion in there that !isRoot in the PushConst case, and not have the InlineBuffer?

Reply:
See other comment -- the driver uses PushConst in a non-root node to mean what I now call InlineBuffer in LGC. That translation is done in LLPC PipelineContext when translating LLPC's user data layout structs to LGC's one.

Reply:
You're right that we do not attempt to partly unspill the push constants block.